udemy_exercises/songs/SongPlayer.java

```diff
@@ -0,0 +1,236 @@
+            sc.nextLine();
+            song = new Song(title, duration);
+            album.addSong(song);
+            showPrompt("Add another song (y/N)?");
+//            answer = getSelection();
+            answer = sc.nextLine();
+            System.out.println(answer);
+            loadMore = answer.equals("y") || answer.equals("yes") ? true : false;
+        } while(loadMore);
+        showPrompt("What's the album title?");
+        album.setName(getSelection());
+        this.albums.add(album);
+        System.out.println("Album added!");
+        displayMainMenu();
+    }
+    private void findAlbum() {
+        String name;
+        showPrompt("Which album would you like to find?");
+        name = getSelection();
+        Album album = this.albums.stream()
+                .filter(a -> {
+                    return a.getName().equals(name);
+                })
+                .findAny()
+                .orElse(null);
+        if(album != null) {
+            System.out.println(album.getName());
+            album.listSongs();
+        } else {
+            System.out.println("Album does not exist!");
+        }
+        displayMainMenu();
+    }
+    private void deleteAlbum() {
+        String name;
+        showPrompt("Which album would you like to delete?");
+        name = getSelection();
+        Album album = this.albums.stream()
+                .filter(a -> {
+                    return a.getName().equals(name);
+                })
+                .findAny()
+                .orElse(null);
+        if(album != null) {
+            this.albums.remove(album);
+            System.out.println("Album removed!");
+        } else {
+            System.out.println("Album does not exist!");
+        }
+        displayMainMenu();
+    }
+    private String getSelection() {
+        String selection = sc.nextLine();
+        return selection;
+    }
+    private void showPrompt(String prompt) {
+        System.out.println(prompt);
+    }
+    private void showPrompt(String[] prompts) {
+        for(String p: prompts) {
+            showPrompt(p);
+        }
+    }
+
+    public void play() {
+        if(this.songs.size() == 0) {
+            System.out.println("Please load some songs first!");
+        } else {
+            System.out.println("Currently playing: " + this.songs.get(this.currentTrack).getTitle());
+        }
+        loadCategoryMenu("Player");
+    }
+    public void replay() {
+        play();
+    }
+    public void next() {
+        this.currentTrack++;
+        play();
+    }
+    public void previous() {
+        this.currentTrack--;
+        play();
+    }
+    public void stop() {
+        System.out.println("Stopped!");
+        loadCategoryMenu("Player");
+    }
+    private void exit() {
+        this.showMainLoop = false;
+        displayMainMenu();
+    }
+    public static void main(String[] args) {
+        SongPlayer sp = new SongPlayer();
+        sp.displayMainMenu();
+    }
```

Comment:
You've encapsulated the 'menu' responsibilities well.  However, this has resulted in a very large class, which overall handles quite a bit, reducing the Cohesion of the class.  Consider which responsibilities may _**not**_ really belong to the 'SongPlayer', and how they might be extracted into a new class to handle those responsibilities specifically.